packages/cli/src/ui/utils/clipboardUtils.ts

```diff
@@ -146,7 +156,91 @@ export async function cleanupOldClipboardImages(
         }
       }
     }
-  } catch {
+  } catch (e) {
     // Ignore errors in cleanup
+    debugLogger.debug('Failed to clean up old clipboard images:', e);
+  }
+}
+
+/**
+ * Splits text into individual path segments, respecting escaped spaces.
+ * Unescaped spaces act as separators between paths, while "\ " is preserved
+ * as part of a filename.
+ *
+ * Example: "/img1.png /path/my\ image.png" → ["/img1.png", "/path/my\ image.png"]
+ *
+ * @param text The text to split
+ * @returns Array of path segments (still escaped)
+ */
+export function splitEscapedPaths(text: string): string[] {
+  const paths: string[] = [];
+  let current = '';
+  let i = 0;
+
+  while (i < text.length) {
+    const char = text[i];
+
+    if (char === '\\' && i + 1 < text.length && text[i + 1] === ' ') {
+      // Escaped space - part of filename, preserve the escape sequence
+      current += '\\ ';
+      i += 2;
+    } else if (char === ' ') {
+      // Unescaped space - path separator
+      if (current.trim()) {
+        paths.push(current.trim());
+      }
+      current = '';
+      i++;
+    } else {
+      current += char;
+      i++;
+    }
+  }
+
+  // Don't forget the last segment
+  if (current.trim()) {
+    paths.push(current.trim());
+  }
+
+  return paths;
+}
+
+/**
+ * Processes pasted text containing file paths, adding @ prefix to valid paths.
+ * Handles both single and multiple space-separated paths.
+ *
+ * @param text The pasted text (potentially space-separated paths)
+ * @param isValidPath Function to validate if a path exists/is valid
+ * @returns Processed string with @ prefixes on valid paths, or null if no valid paths
+ */
+export function processPastedPaths(
```

Comment:
optional nit: processPastedPaths is a bit of a confusing name. Some alternatives:
extractPastedPaths, parsePastedPaths, parseValidPastedPaths